adapters/zap/zap_test.go

```diff
@@ -0,0 +1,70 @@
+package zap_test
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	adapter "github.com/axiomhq/axiom-go/adapters/zap"
+	"github.com/axiomhq/axiom-go/axiom"
+)
+
+func TestCore(t *testing.T) {
+	now := time.Now()
+
+	exp := fmt.Sprintf(`{"_time":"%s","level":"info","key":"value","msg":"my message"}`,
+		now.Format(time.RFC3339Nano))
+
+	hasRun := false
+	hf := func(w http.ResponseWriter, r *http.Request) {
+		gzr, err := gzip.NewReader(r.Body)
+		require.NoError(t, err)
+
+		b, err := ioutil.ReadAll(gzr)
+		assert.NoError(t, err)
+
+		assert.JSONEq(t, exp, string(b))
+
+		hasRun = true
```

Comment:
Axiom expects a status response, that's why you get `EOF`

Reply:
Appreciated!